code/modules/materials/_materials.dm

```diff
@@ -428,7 +428,7 @@ INITIALIZE_IMMEDIATE(/obj/effect/gas_overlay)
 
 //Clausius–Clapeyron relation
 /decl/material/proc/get_boiling_temp(var/pressure = ONE_ATMOSPHERE)
-	return (1 / (1/max(boiling_point, TCMB)) - ((R_IDEAL_GAS_EQUATION * log(pressure / ONE_ATMOSPHERE)) / (latent_heat * molar_mass)))
+	return (1 / (1/max(boiling_point, TCMB)) - ((R_IDEAL_GAS_EQUATION * ((pressure > 0)? log(pressure / ONE_ATMOSPHERE) : 0)) / (latent_heat * molar_mass)))
```

Comment:
it's called with a pressure of 0. Which is legal and to be expected. Except ``log(0)`` is undefined and crashes the proc.

Reply:
Well, the thing is the equation you got there is wrong. Because, let's take water for example :
```
B = 1 / 1 / 100
B = 1 / 0.01
B = 100
```
Sooo, it would mean the boiling point at one atmosphere is the boiling point at 0 atmosphere, which is wrong?
I wonder if there isn't a problem with the whole equation itself?

Reply:
@NataKilar Gotta ask the resident physics man.

Reply:
I think there may be an order of operations and units error, but it otherwise seems correct. I would say:
```
if(pressure == 0)
    return melting_point

p_ratio = log(ONE_ATMOSPHERE / pressure)
molar_heat = latent_heat * molar_mass * 1000 // kJ -> J

. = 1/((R_IDEAL_GAS_EQUATION/molar_heat)*p_ratio + 1/boiling_point)

return max(melting_point, .)
```

At some point between atmospheric pressure and 0 pressure the assumptions of the relation are probably violated somehow, but it's a fine approximation. It doesn't make sense to return a boiling point lower than the melting point.

![image](https://github.com/NebulaSS13/Nebula/assets/20494662/af3a259a-86de-492a-b934-4da7a6c95dd0)

Reply:
The weird thing is looking at some of the values they get from trying to boil water in vacuum, it doesn't add up:
![image](https://github.com/NebulaSS13/Nebula/assets/8772299/f578c01d-6dd6-4ec2-a446-8e316066da31)
They get -68c as boiling point in almost complete vaccum : https://www.engineeringtoolbox.com/water-evacuation-pressure-temperature-d_1686.html

Reply:
What do you mean that doesn't add up? I'm pretty sure it does.

Reply:
I meant, the whole thing about the boiling temp not going lower than the melting temp.

Reply:
But, like nata said on discord, another big problem is we don't calculate the melting point based on temperature and pressure, which will cause problems. That probably should get handled in another PR though.